### PR TITLE
chore: use manageGrpcClient method in Service/init.go

### PR DESCRIPTION
**Pull request by gatici**
All the GRPC client management implementations are done under service/Initialize method in all NFs.
Besides, a new version of config5g library is required.

### CODE REVIEW

File: service/init.go [hunk 1]
```diff
@@ -88,9 +90,66 @@ func (nrf *NRF) Initialize(c *cli.Context) error {
...
+				configChannel = client.PublishOnConfigChange(true, stream)
+				logger.InitLog.Infoln("PublishOnConfigChange is triggered.")
+				go factory.NrfConfig.UpdateConfig(configChannel)
+				logger.InitLog.Infoln("NRF updateConfig is triggered.")
+			}
+		} else {
+			client, err = grpcClient.ConnectToConfigServer(webuiUri)
+			stream = nil
+			configChannel = nil
+			logger.InitLog.Infoln("Connecting to config server.")
```
[the remark this answers is not in the document]
Comment (gatici) [reply service/init.go]: Done

File: service/init.go [hunk 2]
```diff
@@ -88,9 +90,66 @@ func (nrf *NRF) Initialize(c *cli.Context) error {
...
+					logger.InitLog.Infof("failing SubscribeToConfigServer: %+v", err)
+					continue
+				}
+			}
+
+			if configChannel == nil {
+				configChannel = client.PublishOnConfigChange(true, stream)
+				logger.InitLog.Infoln("PublishOnConfigChange is triggered.")
+				go factory.NrfConfig.UpdateConfig(configChannel)
+				logger.InitLog.Infoln("NRF updateConfig is triggered.")
```
[the remark this answers is not in the document]
Comment (gatici) [reply service/init.go]: Done

File: service/init.go [hunk 3]
```diff
@@ -88,9 +90,66 @@ func (nrf *NRF) Initialize(c *cli.Context) error {
...
+				stream, err = client.SubscribeToConfigServer()
+				if err != nil {
+					logger.InitLog.Infof("failing SubscribeToConfigServer: %+v", err)
+					continue
+				}
+			}
+
+			if configChannel == nil {
+				configChannel = client.PublishOnConfigChange(true, stream)
+				logger.InitLog.Infoln("PublishOnConfigChange is triggered.")
```
[the remark this answers is not in the document]
Comment (gatici) [reply service/init.go]: Done

**Comment by gatici** [comment]
The implementations are done under `service/init.go Initialize` method in all the NF's.

File: factory/factory.go [hunk 4]
```diff
@@ -45,55 +42,11 @@ func InitConfigFactory(f string) error {
 		logger.InitLog.Infof("DefaultPlmnId Mnc %v, Mcc %v", NrfConfig.Configuration.DefaultPlmnId.Mnc, NrfConfig.Configuration.DefaultPlmnId.Mcc)
 		if os.Getenv("MANAGED_BY_CONFIG_POD") == "true" {
 			logger.InitLog.Infoln("MANAGED_BY_CONFIG_POD is true")
-			client, err := grpcClient.ConnectToConfigServer(NrfConfig.Configuration.WebuiUri)
-			if err != nil {
-				go updateConfig(client)
-			}
-			return err
 		}
```
Comment (gab-arrobo) [inline factory/factory.go:45]: Why do you need to keep this `if` statement?

**Review by gab-arrobo — APPROVED** [review]
+1

**Pull request merged**